gcp/live/dev/spec/network_spec.rb

```diff
@@ -0,0 +1,168 @@
+require 'json'
+require 'rspec/expectations'
+
+class Pod
+  def self.from_hash(hash)
+    Pod.new(hash)
+  end
+
+  def initialize(hash)
+    @data = hash
+  end
+
+  def ip
+    @data['status']['podIP']
+  end
+
+  def name
+    @data['metadata']['name']
+  end
+
+  def has_label(key, value)
+    @data['metadata']['labels'].keys.include?(key) &&
+    @data['metadata']['labels'][key] == value
+  end
+end
+
+def get_gpii_pods
+  JSON.parse(`kubectl get pods -n gpii -o json`)['items']
+    .collect{ |item| Pod.from_hash(item) }
+end
+
+def get_gpii_pods_labeled(key, value)
+  get_gpii_pods
+    .select{ |item| item.has_label(key, value) }
+end
+
+def kubectl(cmd)
+  `kubectl #{cmd} 2>/dev/null`
+end
+
+describe "Security:" do
+  before :all do
+    @couchdb_pods = get_gpii_pods_labeled("app", "couchdb")
+    @preferences_pods = get_gpii_pods_labeled("app", "preferences")
+    @flowmanager_pods = get_gpii_pods_labeled("app", "flowmanager")
+    @all_pods = @couchdb_pods | @preferences_pods | @flowmanager_pods
+
+    @couchdb_pods.each do |pod|
+      kubectl("exec -n gpii -it #{pod.name} -- sh -c \"[ ! -f /bin/nc ] && apt-get update -y && apt-get install netcat -y\"")
```

Comment:
We should definitely not install anything into or modify running containers, the less nc. Actually, a good practice that K8s support is to run containers with read-only fs.

Reply:
As an alternative, how about downloading `busybox` into `/tmp` and running `netcat` from there?

Reply:
Usually at least `/tmp` needs to be writeable right? Also, do we have a security requirement to make the containers read-only? If not, I'd rather not over-engineer our security tests. I would argue that If we execute these tests in a disposable environment (not in stg or prd), then we wouldn't be increasing our attack surface at all.